core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/DiagnosticLog.java

```diff
@@ -102,4 +128,208 @@ public Map<String, Object> getConfigurations() {
+         */
+        public DiagnosticLogBuilder putParams(String key, Object value) {
+
+            if (this.input == null) {
+                this.input = new HashMap<>();
+            }
+            this.input.put(key, value);
+            return this;
+        }
+
+        /**
+         * Adds a new key-value pair to the configurations map of the DiagnosticLog.
+         * If the configurations map has not been initialized yet, this method will initialize it.
+         *
+         * @param key   the key to be added to the configurations map.
+         * @param value the value to be associated with the key in the configurations map.
+         * @return the current DiagnosticLogBuilder instance.
+         */
+        public DiagnosticLogBuilder putConfigurations(String key, Object value) {
+
+            if (this.configurations == null) {
+                this.configurations = new HashMap<>();
+            }
+            this.configurations.put(key, value);
+            return this;
+        }
+
+        /**
+         * Adds all entries from the provided map to the input map of the DiagnosticLog.
+         * If the input map has not been initialized yet, this method will initialize it.
+         *
+         * @param input the map containing entries to be added to the input map.
+         * @return the current DiagnosticLogBuilder instance.
+         */
+        public DiagnosticLogBuilder addAllParams(Map<String, Object> input) {
+
+            if (this.input == null) {
+                this.input = new HashMap<>();
+            }
+            this.input.putAll(input);
+            return this;
+        }
+
+        /**
+         * Adds all entries from the provided map to the configurations map of the DiagnosticLog.
+         * If the configurations map has not been initialized yet, this method will initialize it.
+         *
+         * @param configurations the map containing entries to be added to the configurations map.
+         * @return the current DiagnosticLogBuilder instance.
+         */
+        public DiagnosticLogBuilder addAllConfigurations(Map<String, Object> configurations) {
+
+            if (this.configurations == null) {
+                this.configurations = new HashMap<>();
+            }
+            this.configurations.putAll(configurations);
+            return this;
+        }
+
+        /**
+         * Sets the log level of the DiagnosticLog.
+         *
+         * @param logLevel the log level to be set.
+         * @return the current DiagnosticLogBuilder instance.
+         */
+        public DiagnosticLogBuilder logLevel(LogLevel logLevel) {
+
+            this.logLevel = logLevel;
+            return this;
+        }
+
+        /**
+         * Constructs a DiagnosticLog object using the set parameters.
+         *
+         * @return a new DiagnosticLog object.
+         */
+        public DiagnosticLog build() {
+
+            if (this.logId != null) {
```

Comment:
not null? shouldn't we add a UUID if it's null?

Reply:
Fixed with b4971ec182001f59d683ea61b29217e4089ab118